index.html

```diff
@@ -0,0 +1,482 @@
+}
+
+type Astronaut
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!     @csdl_resolve(graph: ASTRONAUTS)
+  name: String!   @csdl_resolve(graph: ASTRONAUTS)
+  tripId: String! @csdl_resolve(graph: ASTRONAUTS)
+  rocket: Rocket! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Rocket 
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!           @csdl_resolve(graph: ROCKETS)
+  name: String!         @csdl_resolve(graph: ROCKETS)
+  astronaut: Astronaut! @csdl_resolve(graph: ASTRONAUTS)
+  captain: Astronaut    @csdl_resolve(graph: ASTRONAUTS,
+                                      provides: "{ tripId }")
+}
+```
+
+The meaning of these directives is explored the [Directives](#directives) section.
+
+## Actors
+
+```mermaid
+#::[Actors and roles within an example composition pipeline]
+flowchart TB
+    subgraph A [subgraph A]
+      schemaA([schema A])
+      endpointA([endpoint A])
+    end    
+    subgraph B [subgraph B]
+      schemaB([schema B])
+      endpointB([endpoint B])
+    end
+    subgraph C [subgraph C]
+      schemaC([schema C])
+      endpointC([endpoint C])
+    end
+    subgraph Producer
+      Composer
+    end
+    CSDL([CSDL])
+    subgraph Consumer
+      Router
+    end    
+    A-->Composer
+    B-->Composer
+    C-->Composer
+    Composer-->CSDL
+    CSDL-->Router
+    Router-->published([Published Schema])
+    published-->Clients
+    Clients-->Router
+```
+
+<a name=def-producer></a>
+**Producers** generate CSDL. This spec places requirements on csdl producers.
+
+<a name=def-consumer></a>
+**Consumers** consume CSDL. This spec places requirements on csdl consumers.
+
+<a name=def-composer></a>
+**Composers** or **compilers** are producers which compose subgraph schemas into CSDL. This document places no particular requirements on the composition algorithm, except that it must produce valid csdl.
+
+<a namme=def-router></a>
+**Routers** are consumers which serve a composed schema as a GraphQL endpoint.
+  - Graph routers differ from standard GraphQL endpoints in that they are not expected to process data or communicate with (non-GraphQL) backend services on their own. Instead, graph routers receive GraphQL requests and service them by performing additional GraphQL requests. This spec provides guidance for implementing routers, but does not require particular implementations of query separation or dispatch.
+  - Routers **may**—and typically will—omit CSDL directives and types from the schema they present to clients via introspection
+
+<a name=def-endpoint></a>
+**Endpoints** are running servers which can resolve GraphQL queries against a schema. In this version of the spec, endpoints must be URLs, typically http/https URLs.
+
+<a name=def-subgraph></a>
+**Subgraphs** are the GraphQL schemas which were composed to form the CSDL. A subgraph has:
```

Comment:
Agree it's all or nothing -- I'm a bit hesitant to fully embrace "subgraph", but I'm more interested in a single word than what that word is.

Reply:
More specifically, it is useful to have a single word that distinguishes between a schema designed to become part of a federation from that which doesn't. The way we're talking about this in the registry is modeling a way for a given "graph" to register a partial schema. Calling that the subgraph schema could make sense. Staying `rover subgraph push` could also make sense.

Reply:
From my perspective, there are two things that deserve terms, and I'll call them what they appear to be now:
**sub-graph** — the entity, from the perspective of a gateway, that exposes a partial schema to be composed with other partial schemas to form the CSDL.
**partial schema** — a schema that is compliant to the federation specification and can be composed with other partial schemas

I think that we should **not** call something a sub-graph when we are indeed referring to a schema. I think it's useful to have a distinction between these two terms. I think that just because an entity is referenced as a sub-graph of a gateway does not preclude it from being a (normal) graph in its own right.

Reply:
subgraphs here are explicitly neither partial nor necessarily "designed to become part of a federation," fwiw.